[den] 2주차/mid

sjuhan123 opened the pull request:
## 진행 상황

### midpoint 구현 feature

- grid view + page 기능 구현
- grid view 구독 및 해지 기능 -> "내가 구독한 언론사" 탭에 반영
- "내가 구독한 언론사" 탭 구현

### final 구현 필요 feature

- list view

## 고민 중인 부분

사용자가 구독한 언론사들에 대한 정보를 계속 보관하고, 지속 활용해야 되는데, 해당 정보를 어떤 데이터 구조에 보관할지와 어디에 보관해야지 효율적일지 고민입니다. 

+ 로직 내에 데이터를 보관하면 사용자가 페이지 새로고침시 데이터도 reset 되는데, 그래서 local storage라는 개념도 활용해볼까 생각 중입니다.

## Week2 학습계획

### Week2: model-view 역할분리

**:one: 월요일 학습계획**

- [x] : 기능 구현
  - [x] : grid hover시 구독하기 버튼 나타나게 구현
  - [x] : 구독하기 버튼 클릭시 해지하기 버튼 구현
  - [x] : 구독하기 버튼 클릭시 '내가 구독한 언론사' 탭에 해당 언론사 추가 기능 구현
  - [x] : 해지하기 버튼 클릭시 '내가 구독한 언론사' 탭에서 해당 언론사 삭제 기능 구현
  - [x] : 구독하기 버튼 연속 클릭시 '내가 구독한 언론사' 탭에 해당 언론사 계속 추가 되지 않도록 차단 기능 구현
  - [x] : '내가 구독한 언론사'탭의 리스트가 24개 이하일 때 버튼 노출되지 않도록 구현
  - [x] : '내가 구독한 언론사'탭에서 다시 '전체 언론사' 탭으로 돌아 왔을 때 각 언론사별 구독 여부 데이터 유지되게끔 구현
- [ ]: 기능 구현 이후 코드 리팩토링 진행

**:two: 화요일 학습계획**

- [x] : 기능 구현
  - [x] : main view header 영역 font style 구현
  - [x] : 자주 쓰이는 css 속성 모아 놓은 variable.css 파일 생성
  - [x] : fix: grid view click 이벤트 오류 해결
- [ ] : scss 개념 학습 및 설치
- [ ] : scss 프로젝트에 적용
  - [ ] : header 영역 적용
  - [ ] : trend 영역 적용
  - [ ] : main view 영역 적용

**:three: 수요일 학습계획**

- [x] : 기능 구현
  - [x] : mainView 버튼 css 구현, 클릭 이벤트 추가
  - [x] : grid item hover시 배경 색깔 변경되도록 구현
  - [x] : main-view 버튼, header영역, trendnews영역 css 구현
  - [x] : 전체언론사, 내가구독한언론사 버튼 클릭시 font style 변경 이벤트 구현
  - [x] : header, trendnews 영역 css, html 초기 구현 진행
- [ ] : store 개념 학습
- [ ] : store 개념 코드에 적용

**:four: 목요일 학습계획**

- [ ] : trend 롤링바 구현

**:five: 금요일 학습계획**

- [ ] : list뷰 구현

sjuhan123 committed:
feat: grid view 구독 기능 구현

1. grid hover시 구독하기 버튼 나타나게 구현
2. 구독하기 버튼 클릭시 해지하기 버튼 구현
3. 구독하기 버튼 클릭스 '내가 구독한 언론사' 탭에 추가 기능 구현
4. 해지하기 버튼 클릭시 '내가 구독한 언론사' 탭에서 삭제 기능 구현
5. 구독하기 버튼 연속시 '내가 구독한 언론사' 탭에 계속 추가 되지 않도록 차단 기능 구현
6. '내가 구독한 언론사'탭의 리스트가 24개 이하일 때 버튼 노출되지 않도록 구현
7. '내가 구독한 언론사'탭에서 다시 '전체 언론사' 탭으로 돌아 왔을 때 각 언론사별 구독 여부 데이터 유지되게끔 구현

sjuhan123 committed:
docs: Week2 주차 학습계획 작성 및 맥북 교체로 인한 세팅 진행

sjuhan123 committed:
fix: grid view click 이벤트 오류 해결

sjuhan123 committed:
feat: 자주 쓰이는 css 속성 모아 놓은 variable.css 파일 생성

sjuhan123 committed:
feat: main view header 영역 클릭 이벤트 추가

전체 언론사,내가 구독한 언론사 버튼 클릭시 색깔변경 효과 기능 구현

sjuhan123 committed:
feat: main view header 영역 font style 구현

sjuhan123 committed:
delete: 미사용 아이콘 이미지 파일 삭제

sjuhan123 committed:
feat: header, trendnews 영역 css, html 초기 구현 진행

sjuhan123 committed:
feat: main-view 버튼, header영역, trendnews영역 css 구현

sjuhan123 committed:
feat: 전체언론사, 내가구독한언론사 버튼 클릭시 font style 변경 이벤트 구현

sjuhan123 committed:
feat: grid item hover시 배경 색깔 변경되도록 구현

sjuhan123 committed:
refactor: grid view page 속성 가져오는 로직 중 하드코딩된 부분 수정

sjuhan123 committed:
feat: mainView view 버튼 컴퍼넌트로 다시 구현

1. view 버튼 css 구현
2. view 버튼 클릭 이벤트 - 클릭시 색깔 상태 변경 구현

sjuhan123 committed:
refactor: app mounted 메서드 내 로직 순서 변경

비동기 처리와 상관없는 코드 비동기 처리 실행 전으로 이동

poco111 commented on src/components/Header.js:
```diff
@@ -2,15 +2,24 @@ import { Component } from "../core/Component.js";
+      day: "numeric",
+      weekday: "long",
+    };
+    const formattedDate = date.toLocaleString("ko-KR", options);
```
toLocaleString 메서드 활용도 인상 깊은데, 두번째 인자 options를 객체화해서 만든 것도 신기하네요!

poco111 commented on src/variables.css:
```diff
@@ -0,0 +1,32 @@
+  font-size: 16px;
+}
+
+:root {
```
css 변수를 상수화 좋습니다!

poco111 commented on src/components/Header.js:
```diff
@@ -2,15 +2,24 @@ import { Component } from "../core/Component.js";
 
 export class Header extends Component {
   setUp() {
```
class 상속과 오버라이딩 활용이 좋습니다! 저도 공부해볼게요!

sjuhan123 committed:
docs: Week2 day2,3 학습계획 작성

sjuhan123 committed:
docs: 주석 추가

crongro reviewed (changes requested):
App.js 만 보면 mounted 가 어떤 역할인지 좀 헷갈립니다. App의 constructor 에서(객체 초기화 단계) 해야할 것 같은 다른 객체 초기화 작업을 mounted 에서 하는것 같기도 해서요.

ViewContainer 디렉토리는 메인뷰에 대한 파일들일까요? 이름만보고 viewConainer 라고 하니 view용 container 모음이라고 생각이 들었네요. 그리고 그 디렉토리 아래에는 view도 있고 view 가 안붙은 파일도 있고 viewcontainer 라는 디렉토리와 동일한 파일도 있고, 조금 정리가 필요합니다. 심지어 어떤 이름은 buttonView가 아니고 viewBtn 이네요.

그러려면 먼저 디렉토리의 역할을 정의하고 그에 맞는 디렉토리 이름을 짓고요.(구조를 표현하기보다는 의미를담은 디렉토리 명)  그 안에 파일도 그 역할에 맞는 이름으로 짓고요. 

addEventlistener의 콜백함수는 분리해서 클래스내에 다른 메서드로 정의해주세요. 특히 복잡한 파일이면 그래야 할 듯.


수고하셨어요. 리뷰 참고해서 전체 구조와 네이밍을 한번 수정해보세요.

sjuhan123 committed:
feat: list view 구현 시작

1. list view에 필요한 데이터 가공  로직 구현
2. list view main html, css 구현
3. list view header 구현 완료
4. list view header css 구현

sjuhan123 committed:
feat: list view main, sub section css 구현

sjuhan123 committed:
feat: view-container 클래스명 변경, pressData suffle 기능 추가

sjuhan123 committed:
feat: list view main 영역 구현

sjuhan123 committed:
fix: 프로젝트 폴더 구조, 전체적인 파일명  수정

sjuhan123 committed:
fix: api, crawl 파일 디렉토리 변경

sjuhan123 committed:
fix: app 내 api import path 변경

crongro commented on src/components/MainView/MainView.js:
```diff
@@ -0,0 +1,111 @@
+    `;
+  }
+
+  mounted() {
```
mounted  기능이  비교적 복잡하네요.
어떤 일을 하는 지 좀더 잘 드러나게 수정해도 좋을 듯.

crongro commented on src/components/MainView/ListView/ListViewHeader.js:
```diff
@@ -0,0 +1,48 @@
+
+    const headCategoryNodes =
+      btnState === "all-press"
+        ? categoryIds.reduce((acc, category) => {
```
이런코드 문법을 많이 쓰긴해요. (템플릿 엔진으로 동작하는 렌더링방식에서)
그래도 reduce 코드가 길면 그 콜백을 분리하는 경우도 있긴하고요.

crongro commented on src/components/MainView/ListView/ListView.js:
```diff
@@ -0,0 +1,122 @@
+    } = this._state;
+
+    new ListViewHeader(listViewHeader, {
+      currentPageInCategory: currentPageInCategory,
+      currentCategory: currentCategory,
+      currentCategoryTotalPage: currentCategoryTotalPage,
```
key-value 가 같으면 생략할 수 있어요~

new ListViewHeader(listViewHeader, {
  currentPageInCategory,
  currentCategory,
  currentCategoryTotalPage
});

crongro reviewed (approved):
수고하셨어요~ 리뷰 좀 남겼어요.
